subsys/logging/log_core.c

```diff
@@ -937,9 +937,7 @@ static int enable_logger(void)
 				K_KERNEL_STACK_SIZEOF(logging_stack),
 				log_process_thread_func, NULL, NULL, NULL,
 				LOG_PROCESS_THREAD_PRIORITY, 0,
-				COND_CODE_1(CONFIG_LOG_PROCESS_THREAD,
-					K_MSEC(CONFIG_LOG_PROCESS_THREAD_STARTUP_DELAY_MS),
-					K_NO_WAIT));
+				K_MSEC(CONFIG_LOG_PROCESS_THREAD_STARTUP_DELAY_MS));
```

Comment:
i think that this shouldn't be touched. There is a case when CONFIG_LOG_PROCESS_THREAD=n and then CONFIG_LOG_PROCESS_THREAD_STARTUP_DELAY_MS does not exists. My bad that there is no test that shows that. I will try to add one.

Reply:
@nordic-krch sorry, don't understand. Please check line 932 above, this code is *only* entered if `CONFIG_LOG_PROCESS_THREAD` is enabled.

Reply:
the benefit of using IS_ENABLED is that there is less #fidefs, if you add ifdef then you also need to add one stack and thread structure declaration thus imo it's better to keep it as is.